services/test_runner_service.rb

```diff
@@ -198,5 +215,24 @@ def update_build_status_source!
       logger.error(ex.to_s)
       logger.error(ex.backtrace.join("\n"))
     end
+
+    def fastfile_path_from_project!(project: nil)
```

Comment:
Why is that code needed? I thought we already detect the Fastfile, because we already have the lane selection

Reply:
The lane detection is made from the `project_controller` itself with exactly the same code defined there, and this has relationship with what I commented to you in slack, let me extend the thought in the comment below.